src/Symfony/Component/Form/Tests/Extension/Core/DataTransformer/DateTimeToLocalizedStringTransformerTest.php

```diff
@@ -141,6 +141,22 @@ public function testTransformWithDifferentPatterns()
         $this->assertEquals('02*2010*03 04|05|06', $transformer->transform($this->dateTime));
     }
 
+    public function testTransformDateTimeImmutableTimezones()
+    {
+        if (PHP_VERSION_ID < 50500) {
+            $this->markTestSkipped('DateTimeImmutable was introduced in PHP 5.5.0');
+        }
+
+        $transformer = new DateTimeToLocalizedStringTransformer('America/New_York', 'Asia/Hong_Kong');
+
+        $input = new \DateTimeImmutable('2010-02-03 04:05:06 America/New_York');
+
+        $dateTime = clone $input;
+        $dateTime = $dateTime->setTimezone(new \DateTimeZone('Asia/Hong_Kong'));
```

Comment:
I think that this can be simplified, you already clone the `$dateTime` inside the `DateTimeToLocalizedStringTransformer`

```php
$dateTime = new \DateTimeImmutable('2010-02-03 04:05:06 America/New_York');
$dateTime->setTimezone(new \DateTimeZone('Asia/Hong_Kong'));
```